source/JsMaterialX/JsMaterialXView/src/index.js

```diff
@@ -186,91 +85,83 @@ function init() {
     var fileloader = new THREE.FileLoader();
     const objLoader = new OBJLoader();
     const hdrloader = new RGBELoader();
+    const textureLoader = new THREE.TextureLoader();
 
     Promise.all([
-        new Promise(resolve => hdrloader.setDataType(THREE.FloatType).load('san_giuseppe_bridge_split.hdr', resolve)),
-        new Promise(resolve => hdrloader.setDataType(THREE.FloatType).load('irradiance/san_giuseppe_bridge_split.hdr', resolve)),
-        new Promise(resolve => objLoader.load('shaderball.obj', resolve)),
-        new Promise(resolve => fileloader.load('shader-frag.glsl', resolve)),
-        new Promise(resolve => fileloader.load('shader-vert.glsl', resolve))
-    ]).then(([loadedRadianceTexture, loadedIrradianceTexture, obj, fShader, vShader]) => {
+        new Promise(resolve => hdrloader.setDataType(THREE.FloatType).load('Lights/san_giuseppe_bridge_split.hdr', resolve)),
+        new Promise(resolve => hdrloader.setDataType(THREE.FloatType).load('Lights/irradiance/san_giuseppe_bridge_split.hdr', resolve)),
+        new Promise(resolve => objLoader.load('Geometry/shaderball.obj', resolve)),
+        new Promise(function (resolve) { 
+          MaterialX().then((module) => { 
+            resolve(module); 
+          }); }),
+        new Promise(resolve => materialFilename ? fileloader.load(materialFilename, resolve) : resolve())
+    ]).then(async ([loadedRadianceTexture, loadedIrradianceTexture, obj, mxIn, mtlxMaterial]) => {
+        // Initialize MaterialX and the shader generation context
+        mx = mxIn;
+        let doc = mx.createDocument();
+        let gen = new mx.EsslShaderGenerator();
+        let genContext = new mx.GenContext(gen);
+        let stdlib = mx.loadStandardLibraries(genContext);
+        doc.importLibrary(stdlib);        
+
+        // Load material
+        if (mtlxMaterial)
+            await mx.readFromXmlString(doc, mtlxMaterial);
+        else
+            fallbackMaterial(mx, doc);
+
+        let elem = mx.findRenderableElement(doc);
+
+        const isTransparent = mx.isTransparentSurface(elem, gen.getTarget());
```

Comment:
`elem` can be a `nullptr`.

Reply:
Yes, but in the context of this simple example I don't want to complicate things with error handling everywhere. We basically assume in the whole app that things work as expected